app/models/question.rb

```diff
@@ -9,5 +25,14 @@ class Question < ApplicationRecord
   has_many :dropdown_options
   has_many :range_options
 
+
   validates :text, :question_type, :parameter, presence: true
+  validate :matches_parent_category
+
+  def matches_parent_category
+    return if parent_option.nil?
+    return if parent_option.question.category == category
```

Comment:
parent_option.question gives you the question it belongs to, not the question that is has as a child. You're comparing the parent question to the child question.

Reply:
I thought that was the point of the validation; this way parent questions always persist in the same category as the children question. This way, when we query questions by category we can get all of the questions and all of their children. If this check wasn't in place, then the question could be taken from the category it is in and also from a pointer from its parent (and its parent would be in a different category), so it would be in two places, which is something I thought we wanted to avoid

Reply:
oh rip me i read it as question type yeah we good

Reply:
when we go to implement the interface we should pre-fill the category of all children questions to match their parent